### ISSUE TITLE
prev_prompt_ids.append(tokenizer(prev_prompt, truncation=True, padding="max_length", max_length=tokenizer.model_max_length, return_tensors="pt").input_ids.squeeze(0))

### ISSUE THREAD

**Issue by lq-blackcat** [haoningwu3639/StoryGen#15]
The line of code prev_prompt_ids.append(tokenizer(prev_prompt, truncation=True, padding="max_length", max_length=tokenizer.model_max_length, return_tensors="pt").input_ids.squeeze(0)) should not include .squeeze(0). The shape of t_prev_prompt_ids in t_prev_prompt_ids = torch.stack(prev_prompt_ids) is (3, 77).

Regarding the difference in memory usage between the first and second stages of training, and the necessity to set the batch size to 1 in order for it to work on a 48GB A6000 GPU.

**Comment by haoningwu3639** [comment]
For the first part, we aim to squeeze the 'batch' dimension, which may vary depending on the way of loading data. We will check the code again later.
And for the second part, in the second stage of training, the model needs to use the visual-language context module to calculate the diffusion features of the previous condition frames, so it may require larger GPU memory, that is, smaller batch size, but we still recommend you to use 'accelerate' that can load 'vae' and 'text_encoder' as 'fp16', which may be helpful for more efficient training.
